src/dmd/dsymbol.d

```diff
@@ -243,6 +243,8 @@ extern (C++) class Dsymbol : RootObject
     {
         if (this == o)
             return true;
+        if (o.dyncast() != DYNCAST.dsymbol)
```

Comment:
Is this extra test necessary? Your new call for namespaces already verifies that a Dsymbol is passed. Or did you see other failures here?

Reply:
It's not strictly necessary no. But when developing and trying out other solutions, I was surprised that the argument was blindly casted, which resulted in some memory corruption. It wouldn't surprise me if it would hit other people in the future.

Reply:
Ok. It seems a similar check exists in about half of the `equals` oerloads, but misses in the other half. This base class is unlikely to be hit that often, so the extra check should be ok.